src/sonarqube-gui-plugin/src/main/resources/static/js/cloneViewer/CloneViewer.js

```diff
@@ -0,0 +1,146 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * The object handling visual representation of the code
+ * @param {[type]} HTMLelem [description]
+ * @param {[type]} options  [description]
+ */
+SM.CloneViewer = function() {
+  this.elem = null; // :DOMElement
+  this.maxInstances = 2; // :int
+  this.cloneClassSelector = null; // :SM.ClassSelector
+  this.cloneInstanceSelectors = null; // :SM.InstanceSelector[]
+  this.codeBrowser = null; // :SM.SideBySideDiffer (manages the diffs)
+
+  this.selectedCloneClass = null;
+  this.selectedInstances = null;
+  this.codeBrowser = null;
+
+  var self = this;
+
+  this.init = function() {
+    this.selectedCloneClass = SM.state[SM.options.component.key].cloneViewer.selectedCloneClass;
+    this.selectedCloneClass = (this.selectedCloneClass)? this.selectedCloneClass : 0;
+    this.selectedInstances = SM.state[SM.options.component.key].cloneViewer.selectedInstances;
+    this.selectedInstances = (this.selectedInstances)? this.selectedInstances : [];
+
+    this.codeBrowser = new SM.SideBySideDiffer($("#cloneViewerConatiner"), {});
+
+    this.cloneClassSelector = new SM.CloneClassSelector(
+      $("#cloneClassSelectorContainer"),
+      {
+        cloneClassList: SM.state[SM.options.component.key].clone.data,
+        selected: this.selectedCloneClass
+      }
+    );
+
+    this.cloneInstanceSelectors = [];
+    for (var i = 0; i < this.maxInstances; i++) {
+      $("#cloneInstanceSelectorContainer").append('<div id="cloneInstanceSelector' + i + '"></div>');
+      var instanceList = this.cloneClassSelector.cloneClassList[this.selectedCloneClass].cloneInstances;
+      this.cloneInstanceSelectors[i] = new SM.CloneInstanceSelector(
+        $("#cloneInstanceSelector" + i),
+        {
+          cloneInstanceList: instanceList,
+          selected: (this.selectedInstances[i])? this.selectedInstances[i] : 0,
+          id: i
+        }
+      );
+      this.cloneInstanceSelectors[i].subscribe("onSelect", this.handleCloneInstanceChange);
+    }
+    this.cloneClassSelector.subscribe("onSelect", this.handleCloneClassChange);
+
+    this.cloneClassSelector.select(this.selectedCloneClass) // triggers events
+  };
+
+  /**
+   * Generates HTML of dependant of the current state of the object.
+   * Then overwrites the HTML in the visible webpage as well.
+   * @return {void}
+   */
+  this.renderAll = function() {
+    this.cloneClassSelector.renderAll();
+    this.cloneInstanceSelectors.renderAll();
+    this.codeBrowser.renderAll();
+  };
+
+  /**
+   * is called by CloneClassSelector
+   * @return {void}
```

Comment:
Please add a new line before `@return` and write `undefined` instead of void like before.